flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/FlinkRexUtil.scala

```diff
@@ -535,6 +534,29 @@ object FlinkRexUtil {
         throw new IllegalArgumentException(s"Unknown expression type '${expr.getClass}': $expr")
     }
   }
+
+  /**
+   * Similar to [[RexUtil#findOperatorCall(SqlOperator, RexNode)]] but for flink's udf.
+   */
+  def hasUdfInvocation(expr: RexNode, functionName: String): Boolean = {
```

Comment:
Maybe pass a `BuiltInFunctionDefinition` here instead of the string?

Reply:
I generalized a bit this function, using the `BuiltInFunctionDefinition` concept here seems a bit odd and too specialized. Check it out now